plugins/vue/src/index.js

```diff
@@ -0,0 +1,112 @@
+import VueTypes from 'vue-types';
+import IMask from 'imask';
+
+VueTypes.sensibleDefaults = false;
+
+const optionsTypes = {
+  // common
+  mask: VueTypes.oneOfType([
+    VueTypes.array,
+    VueTypes.func,
+    VueTypes.string,
+    VueTypes.instanceOf(RegExp),
+    VueTypes.oneOf([Date, Number]),
+    VueTypes.instanceOf(IMask.Masked),
+  ]).isRequired,
+  value: VueTypes.string,
+  unmaskedValue: VueTypes.string,
+  prepare: VueTypes.func,
+  validate: VueTypes.func,
+  commit: VueTypes.func,
+
+  // events
+  onAccept: VueTypes.func,
+  onComplete: VueTypes.func,
+
+  // pattern
+  placeholderChar: VueTypes.string,
+  placeholderLazy: VueTypes.bool,
+  definitions: VueTypes.object,
+  groups: VueTypes.object,
+
+  // date
+  pattern: VueTypes.string,
+  format: VueTypes.func,
+  parse: VueTypes.func,
+
+  // number
+  radix: VueTypes.string,
+  thousandsSeparator: VueTypes.string,
+  mapToRadix: VueTypes.arrayOf(VueTypes.string),
+  scale: VueTypes.number,
+  signed: VueTypes.bool,
+  normalizeZeros: VueTypes.bool,
+  padFractionalZeros: VueTypes.bool,
+
+  // dynamic
+  dispatch: VueTypes.func
+};
+
+const component = {
+  name: 'MaskedInput',
+  template: '<input>',
```

Comment:
This will not work with Vue runtime build.
https://github.com/vuejs/vue/tree/dev/dist#runtime--compiler-vs-runtime-only